### PR TITLE
[docs] Tablet metadata and ASH updates

### PR DESCRIPTION
**Pull request by aishwarya24**
DOC-423

Deploy previews:
https://deploy-preview-23221--infallible-bardeen-164bc9.netlify.app/preview/explore/observability/yb-local-tablets/
https://deploy-preview-23221--infallible-bardeen-164bc9.netlify.app/preview/explore/observability/active-session-history/

### CODE REVIEW

File: docs/content/preview/explore/observability/yb-local-tablets.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ---
-title: View YSQL/YCQL and tablet metadata with yb_local_tablets
-linkTitle: YQL and tablet metadata
-description: View tablet metadata for YSQL and YCQL statements, and system tablets executed by a server.
+title: View metadata for YSQL/YCQL/system tablets on a server
```
Comment (abhinab-yb) [inline docs/content/preview/explore/observability/yb-local-tablets.md:2]: The title change is not being reflected in the deployment

File: docs/content/preview/explore/observability/active-session-history.md [hunk 2]
```diff
@@ -31,28 +31,27 @@ To use ASH, enable and configure the following flags for each node of your clust
 
 | Flag | Description |
 | :--- | :---------- |
-| allowed_preview_flags_csv | Pass the flags `ysql_yb_ash_enable_infra` and `ysql_yb_enable_ash` in this flag in CSV format. |
+| allowed_preview_flags_csv | Set the value of this flag to include `ysql_yb_ash_enable_infra` and `ysql_yb_enable_ash`. |
```
Comment (abhinab-yb) [inline docs/content/preview/explore/observability/active-session-history.md:34]:
```suggestion
| allowed_preview_flags_csv | Set the value of this flag to include `ysql_yb_ash_enable_infra`,`ysql_yb_enable_ash`. |
```

While setting the flag, we would need a comma, that's why there was a comma here

File: docs/content/preview/explore/observability/active-session-history.md [hunk 3]
```diff
@@ -66,30 +65,30 @@ ASH exposes the following views in each node to analyze and troubleshoot perform
...
+This view provides a list of wait events and their metadata. The columns of the view are described in the following table.
 
 | Column | Type | Description |
 | :----- | :--- | :---------- |
 | root_request_id | UUID | A 16-byte UUID that is generated per request. Generated for queries at YSQL/YCQL layer. |
 | rpc_request_id | integer | ID for internal requests, it is a monotonically increasing number for the lifetime of a YB-TServer. |
-| wait_event_component | text | There are three components: YSQL, YCQL, and YB-TServer. |
-| wait_event_class | text | Every wait event has a class associated with it. |
-| wait_event | text | Provides insight into what the RPC is waiting on. |
+| wait_event_component | text | Components of the wait event which can be YSQL, YCQL, and TServer. |
```
Comment (abhinab-yb) [inline docs/content/preview/explore/observability/active-session-history.md:74]:
```suggestion
| wait_event_component | text | Component of the wait event which can be YSQL, YCQL, and TServer. |
```

File: docs/content/preview/explore/observability/active-session-history.md [hunk 4]
```diff
@@ -66,30 +65,30 @@ ASH exposes the following views in each node to analyze and troubleshoot perform
...
+| wait_event_class | text | Classes of the wait event such as TabletWait, RocksDB, and so on.  |
```
Comment (abhinab-yb) [inline docs/content/preview/explore/observability/active-session-history.md:75]:
```suggestion
| wait_event_class | text | Class of the wait event such as TabletWait, RocksDB, and so on.  |
```

File: docs/content/preview/explore/observability/active-session-history.md [hunk 5]
```diff
@@ -66,30 +65,30 @@ ASH exposes the following views in each node to analyze and troubleshoot perform
...
 | ysql_session_id | bigint | YSQL session identifier. Zero for YCQL and background activities. |
 | client_node_ip | text | IP address of the client which sent the query to YSQL/YCQL. Null for background activities. |
-| sample_weight | float | If in any sampling interval there are too many events, YugabyteDB only collects `yb_ash_sample_size` samples/events. Based on how many were sampled, weights are assigned to the collected events. <br><br>For example, if there are 200 events, but only 100 events are collected, each of the collected samples will have a weight of (200 / 100) = 2.0 |
+| sample_weight | float | If in any sampling interval there are too many events, YugabyteDB only collects `ysql_yb_ash_sample_size` samples/events. Based on how many were sampled, weights are assigned to the collected events. <br><br>For example, if there are 200 events, but only 100 events are collected, each of the collected samples will have a weight of (200 / 100) = 2.0 |
 
 ## Wait events
 
 List of wait events by the following request types.
```
Comment (abhinab-yb) [inline docs/content/preview/explore/observability/active-session-history.md:87]:
```suggestion
List of wait events by the wait event components.
```

File: docs/content/preview/explore/observability/yb-local-tablets.md [hunk 6]
```diff
@@ -11,9 +11,9 @@ menu:
...
+Use YSQL `yb_local_tablets` view to fetch the metadata for [YSQL](../../../api/ysql/), [YCQL](../../../api/ycql/), and system [tablets](../../../architecture/key-concepts/#tablet) of a node. This view returns the same information that is available on `<yb-tserver-ip>:9000/tablets`.
 
-This view is also present on `<yb-tserver_ip>:9000/tablets` in the form of a YSQL view, and can be joined with YCQL wait events in the [yb_active_session_history](../../observability/active-session-history/#yb-active-session-history) view using the tablet ID. For example, see [Detect a hot shard](../../observability/active-session-history/#detect-a-hot-shard).
+While debugging a system with [Active Session History](../../observability/active-session-history/), the `tablet_id` column of this view can be can be joined with the `wait_event_aux` column of the [yb_active_session_history](../../observability/active-session-history/#yb-active-session-history) view. For example, see [Detect a hot shard](../../observability/active-session-history/#detect-a-hot-shard).
```
Comment (ddhodge) [inline docs/content/preview/explore/observability/yb-local-tablets.md:16]:
```suggestion
While debugging a system with [Active Session History](../active-session-history/), the `tablet_id` column of this view can be can be joined with the `wait_event_aux` column of the [yb_active_session_history](../active-session-history/#yb-active-session-history) view. For example, see [Detect a hot shard](../active-session-history/#detect-a-hot-shard).
```

File: docs/content/preview/explore/observability/active-session-history.md [hunk 7]
```diff
@@ -66,30 +65,30 @@ ASH exposes the following views in each node to analyze and troubleshoot perform
...
 
 | Column | Type | Description |
 | :----- | :--- | :---------- |
 | root_request_id | UUID | A 16-byte UUID that is generated per request. Generated for queries at YSQL/YCQL layer. |
 | rpc_request_id | integer | ID for internal requests, it is a monotonically increasing number for the lifetime of a YB-TServer. |
-| wait_event_component | text | There are three components: YSQL, YCQL, and YB-TServer. |
-| wait_event_class | text | Every wait event has a class associated with it. |
-| wait_event | text | Provides insight into what the RPC is waiting on. |
+| wait_event_component | text | Component of the wait event which can be YSQL, YCQL, and TServer. |
+| wait_event_class | text | Class of the wait event such as TabletWait, RocksDB, and so on.  |
```
Comment (ddhodge) [inline docs/content/preview/explore/observability/active-session-history.md:75]:
```suggestion
| wait_event_class | text | Class of the wait event, such as TabletWait, RocksDB, and so on.  |
```

File: docs/content/preview/explore/observability/active-session-history.md [hunk 8]
```diff
@@ -66,30 +65,30 @@ ASH exposes the following views in each node to analyze and troubleshoot perform
...
+This view provides a list of wait events and their metadata. The columns of the view are described in the following table.
 
 | Column | Type | Description |
 | :----- | :--- | :---------- |
 | root_request_id | UUID | A 16-byte UUID that is generated per request. Generated for queries at YSQL/YCQL layer. |
 | rpc_request_id | integer | ID for internal requests, it is a monotonically increasing number for the lifetime of a YB-TServer. |
-| wait_event_component | text | There are three components: YSQL, YCQL, and YB-TServer. |
-| wait_event_class | text | Every wait event has a class associated with it. |
-| wait_event | text | Provides insight into what the RPC is waiting on. |
+| wait_event_component | text | Component of the wait event which can be YSQL, YCQL, and TServer. |
```
Comment (ddhodge) [inline docs/content/preview/explore/observability/active-session-history.md:74]:
```suggestion
| wait_event_component | text | Component of the wait event, which can be YSQL, YCQL, and TServer. |
```

File: docs/content/preview/explore/observability/active-session-history.md [hunk 9]
```diff
@@ -31,28 +31,27 @@ To use ASH, enable and configure the following flags for each node of your clust
 
 | Flag | Description |
 | :--- | :---------- |
-| allowed_preview_flags_csv | Pass the flags `ysql_yb_ash_enable_infra` and `ysql_yb_enable_ash` in this flag in CSV format. |
+| allowed_preview_flags_csv | Set the value of this flag to include `ysql_yb_ash_enable_infra`, `ysql_yb_enable_ash`. |
 | ysql_yb_ash_enable_infra | Enable or disable ASH infrastructure. <br>Default: false. Changing this flag requires a VM restart. |
-| ysql_yb_enable_ash | Works only in conjunction with the flag `ysql_yb_ash_enable_infra`. Start sampling and instrumentation (that is, periodically check and keep track) of YSQL and YCQL queries, and YB-TServer requests.<br> Default: false. Changing this flag doesn't require a VM restart. |
+| ysql_yb_enable_ash | Works only in conjunction with the flag `ysql_yb_ash_enable_infra`. Setting this flag to true enables collecting of wait events for YSQL and YCQL queries, and YB-TServer requests.<br> Default: false. Changing this flag doesn't require a VM restart. |
```
Comment (ddhodge) [inline docs/content/preview/explore/observability/active-session-history.md:36]:
```suggestion
| ysql_yb_enable_ash | Works only in conjunction with the flag `ysql_yb_ash_enable_infra`. Setting this flag to true enables the collection of wait events for YSQL and YCQL queries, and YB-TServer requests.<br> Default: false. Changing this flag doesn't require a VM restart. |
```

**Review by ddhodge — APPROVED** [review]
comments apply to stable also

File: docs/content/preview/explore/observability/active-session-history.md [hunk 10]
```diff
@@ -31,28 +31,27 @@ To use ASH, enable and configure the following flags for each node of your clust
 
 | Flag | Description |
 | :--- | :---------- |
-| allowed_preview_flags_csv | Pass the flags `ysql_yb_ash_enable_infra` and `ysql_yb_enable_ash` in this flag in CSV format. |
+| allowed_preview_flags_csv | Set the value of this flag to include `ysql_yb_ash_enable_infra`, `ysql_yb_enable_ash`. |
```
Comment (hbhanawat) [inline docs/content/preview/explore/observability/active-session-history.md:34]: `ysql_yb_ash_enable_infra`, `ysql_yb_enable_ash` --> `ysql_yb_ash_enable_infra,ysql_yb_enable_ash`

File: docs/content/preview/explore/observability/yb-local-tablets.md (hunk 6 above)
Comment (hbhanawat) [reply docs/content/preview/explore/observability/yb-local-tablets.md:16]: the `tablet_id` column of this view can be can be joined with --> the `tablet_id` column of this view can be joined with

**Comment by hbhanawat** [comment]
@aishwarya24 @ddhodge Why are we making these changes for stable releases? We have not yet EA'ed the feature.

**Comment by ddhodge** [comment]
> @aishwarya24 @ddhodge Why are we making these changes for stable releases? We have not yet EA'ed the feature.

@hbhanawat I don't know - what versions does it apply to? Ideally there should be a doc ticket for the issue and it would include these details

**Comment by hbhanawat** [comment]
@ddhodge It applies only to 2.21.0.0 release. The initial documentation was done with a doc ticket. https://docs.google.com/document/d/1VRiPYWTclpxBnyxxix9yD87WIZQrsALviDY5qZXha9Y/edit#heading=h.rzphqx7mj8ze

File: docs/content/preview/explore/observability/active-session-history.md [hunk 11]
```diff
@@ -66,30 +65,31 @@ ASH exposes the following views in each node to analyze and troubleshoot perform
...
 
 | Column | Type | Description |
 | :----- | :--- | :---------- |
+| sample_time | timestamp | Timestamp when the sample was captured |
 | root_request_id | UUID | A 16-byte UUID that is generated per request. Generated for queries at YSQL/YCQL layer. |
 | rpc_request_id | integer | ID for internal requests, it is a monotonically increasing number for the lifetime of a YB-TServer. |
-| wait_event_component | text | There are three components: YSQL, YCQL, and YB-TServer. |
-| wait_event_class | text | Every wait event has a class associated with it. |
-| wait_event | text | Provides insight into what the RPC is waiting on. |
+| wait_event_component | text | Component of the wait event, which can be YSQL, YCQL, and TServer. |
```
Comment (abhinab-yb) [inline docs/content/preview/explore/observability/active-session-history.md:75]:
```suggestion
| wait_event_component | text | Component of the wait event, which can be YSQL, YCQL, or TServer. |
```

File: docs/content/preview/explore/query-1-performance/ycql-stat-statements.md [hunk 12]
```diff
@@ -16,19 +17,19 @@ type: docs
 {{<tabitem href="../ycql-stat-statements/" text="YCQL" icon="cassandra" active="true" >}}
 {{</tabs>}}
 
-Databases can be resource-intensive, consuming a lot of memory, CPU, IO, and network resources. By optimizing your CQL, you can minimize resource use. The `ycql_stat_statements` module helps you track planning and execution statistics for all the YCQL statements executed by a server.
+Databases can be resource-intensive, consuming a lot of memory, CPU, IO, and network resources. By optimizing your CQL, you can minimize resource use. The `ycql_stat_statements` module helps you track execution statistics for all the YCQL statements executed by a server.
```
Comment (abhinab-yb) [inline docs/content/preview/explore/query-1-performance/ycql-stat-statements.md:20]:
```suggestion
Databases can be resource-intensive, consuming a lot of memory, CPU, IO, and network resources. By optimizing your YCQL, you can minimize resource use. The `ycql_stat_statements` module helps you track execution statistics for all the YCQL statements executed by a server.
```

File: docs/content/preview/explore/query-1-performance/ycql-stat-statements.md [hunk 13]
```diff
@@ -16,19 +17,19 @@ type: docs
...
+This view is added in a YSQL extension `yb_ycql_utils`, which is not enabled by default.
 
 The columns of the `ycql_stat_statements` view are described in the following table.
 
 | Column | Type | Description |
 | :----- | :--- | :---------- |
 | queryid | int8 | Hash code to identify identical normalized queries. |
 | query | text | Text of a representative statement. |
-| is_prepared  | bool | Indicates whether the statement is a prepared statement or an unprepared query. |
+| is_prepared  | bool | Indicates whether the query statement is prepared statement or unprepared. |
```
Comment (abhinab-yb) [inline docs/content/preview/explore/query-1-performance/ycql-stat-statements.md:32]:
```suggestion
| is_prepared  | bool | Indicates whether the query statement is prepared or unprepared. |
```

**Review by abhinab-yb — APPROVED** [review]

**Pull request merged**